### PR TITLE
Shots and feature map exposure

### PR DESCRIPTION
**Pull request by gcattan**
Expose the number of shots and the feature map (part of issue #13)

### CODE REVIEW

**Review by sylvchev — APPROVED** [review]
Nice work!

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
LGTM! Only minor comments.

File: pyriemann_qiskit/utils/hyper_params_factory.py
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py:14]: This is a union between a string and list of indices. I have now improved the description.

File: pyriemann_qiskit/utils/hyper_params_factory.py
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py]: I wanted to say that the value has to be greater or equal to 1. 
I have now updated the description to put it more clear.

**Comment by gcattan** [comment]
Thank you for the comments. There is still some problem with the pipeline. I should investigate it deeper, but it seems that `git` failed to be installed on the doc pipeline. If you have an idea do not hesitate :)

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
Again, some minor remarks.

More generally:

1 - don't merge master in your branch, but **rebase** your branch on master; otherwise, it is really difficult to understand if modifications com from branch and from master.

2 - to easily update your fork master on the pyRiemann-qiskit master, do:
`git remote add upstream https://github.com/pyriemann/pyriemann-qiskit.git`
Then, before creating a branch, pull from `upstream`.

See SciPy guidelines: https://docs.scipy.org/doc/scipy/reference/dev/contributor/development_workflow.html

**Comment by qbarthelemy** [comment]
> There is still some problem with the pipeline. I should investigate it deeper, but it seems that git failed to be installed on the doc pipeline. If you have an idea do not hesitate :)

I think that the problem comes from module `datasets` of pyRiemann, not declared in init file. Its import fails. This should be corrected after merge of
https://github.com/pyRiemann/pyRiemann/pull/149/commits/91c62e1c6d2c7ad789c186c65ca39a769d33016f
@sylvchev

File: pyriemann_qiskit/utils/hyper_params_factory.py
Comment (gcattan) [reply pyriemann_qiskit/utils/hyper_params_factory.py:4]: Thanks! I added a couple of tests and removed the `pairwise` option which was not available in this version.

File: tests/test_utils_hyper_params_factory.py
Comment (qbarthelemy) [inline tests/test_utils_hyper_params_factory.py:41]: To test if a function raises an error, you can use:
```
with pytest.raises(ValueError):
    my_function.do_a_stuff_raising_an_error()
```

File: tests/conftest.py
Comment (qbarthelemy) [inline tests/conftest.py:70]: For consistency in notations, you should use `n_features` instead of `feature_dim`.

**Review by qbarthelemy — CHANGES_REQUESTED** [review]
Great job!
Last comments, then it will be ok for me.

**Comment by gcattan** [comment]
Thank you for the useful comments! There is still some CI/CD error. I will wait for the aforementioned commit to be merged and come back to this if required :)

**Review by qbarthelemy — APPROVED** [review]
Thx @gcattan !

**Pull request merged**
